paddle/fluid/inference/tensorrt/plugin/generic_plugin.cu

```diff
@@ -529,26 +525,17 @@ int GenericPlugin::enqueue(const nvinfer1::PluginTensorDesc* input_desc,
                            cudaStream_t stream) TRT_NOEXCEPT {
   platform::CUDAPlace place(platform::GetCurrentDeviceId());
   // TODO(inference): generic plugin do not support INT8 precision now.
-  auto protoType2PhiType =
-      [&](GeneratePluginDataType proto_type,
-          nvinfer1::DataType nv_dtype) -> std::pair<phi::DataType, int> {
-    if (proto_type == GeneratePluginDataType::PLUGIN_FP16) {
-      return {phi::DataType::FLOAT16, sizeof(half)};
-    } else if (proto_type == GeneratePluginDataType::PLUGIN_FP32) {
-      if (isFp16Supported() && nv_dtype == nvinfer1::DataType::kHALF) {
-        return {phi::DataType::FLOAT16, sizeof(half)};
-      } else {
-        return {phi::DataType::FLOAT32, sizeof(float)};
-      }
-    } else if (proto_type == GeneratePluginDataType::PLUGIN_INT64) {
-      return {phi::DataType::INT64, sizeof(int64_t)};
-    } else if (proto_type == GeneratePluginDataType::PLUGIN_INT32) {
-      return {phi::DataType::INT32, sizeof(int32_t)};
-    } else if (proto_type == GeneratePluginDataType::PLUGIN_BOOL) {
-      return {phi::DataType::BOOL, sizeof(bool)};
-    } else {
-      CHECK(false) << "precision is not supported";
-    }
+  auto nvType2PhiType =
+      [&](nvinfer1::DataType nv_dtype) -> std::pair<phi::DataType, int> {
+    const std::map<nvinfer1::DataType, std::pair<phi::DataType, int>> _map{
```

Comment:
why remove the mapping of phi::DataType::INT64？

Reply:
TRT does not natively support INT64 type, see [[doc](https://docs.nvidia.com/deeplearning/tensorrt/api/c_api/namespacenvinfer1.html#a83aed11a1c160f30dcd13809678bdd29)] . If Paddle dtype is `phi::DataType::INT64`, it will be internally downcast to INT32, before fed into TRT plugin.